diff --git a/rs/private/rustc_repository.bzl b/rs/private/rustc_repository.bzl
index 9872b34..6c376b6 100644
--- a/rs/private/rustc_repository.bzl
+++ b/rs/private/rustc_repository.bzl
@@ -28,7 +28,7 @@ def _rustc_repository_impl(rctx):
     exec_triple = triple(rctx.attr.triple)
     download_and_extract(rctx, "rustc", "rustc", exec_triple)
     _symlink_rust_objcopy_shared_libraries(rctx, exec_triple)
-    build_content = [BUILD_for_compiler(exec_triple, include_objcopy = True)]
+    build_content = [BUILD_for_compiler(exec_triple, include_linker = True, include_objcopy = True)]
     if includes_rust_analyzer_proc_macro_srv(rctx.attr.version, rctx.attr.iso_date):
         build_content.append(BUILD_for_rust_analyzer_proc_macro_srv(exec_triple))
     rctx.file("BUILD.bazel", "\n".join(build_content))
diff --git a/rs/toolchains/declare_rustc_toolchains.bzl b/rs/toolchains/declare_rustc_toolchains.bzl
index 9d049d1..f8697e7 100644
--- a/rs/toolchains/declare_rustc_toolchains.bzl
+++ b/rs/toolchains/declare_rustc_toolchains.bzl
@@ -59,6 +59,8 @@ def declare_rustc_toolchains(
             rustc = "{}rustc".format(rustc_repo_label),
             cargo = "{}cargo".format(cargo_repo_label),
             clippy_driver = "{}clippy_driver_bin".format(clippy_repo_label),
+            linker = "{}rust-lld".format(rustc_repo_label),
+            linker_type = "direct",
             cargo_clippy = "{}cargo_clippy_bin".format(clippy_repo_label),
             llvm_cov = "@llvm//tools:llvm-cov",
             llvm_profdata = "@llvm//tools:llvm-profdata",
